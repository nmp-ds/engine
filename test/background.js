import { setup } from './_helpers.js';
import { expect, test } from 'vitest';
import { positionMap, globalKeywords } from '#utils';

setup();

test('bg positions', async ({ uno }) => {
  const positions = Object.values(positionMap);
  const classes = positions.map(e => `bg-${e}`);
  const twoDimensionClasses = positions.flatMap(e => positions.map(f => `bg-${e}-${f}`));
  const { css } = await uno.generate([...classes, ...twoDimensionClasses]);
  expect(css).toMatchSnapshot();
});

test('bg repeats', async ({ uno }) => {
  const classes = ['repeat', 'no-repeat', 'repeat-x', 'repeat-y', 'repeat-round', 'repeat-space'].map(e => `bg-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('bg origins', async ({ uno }) => {
  const classes = ['border', 'padding', 'content'].map(e => `bg-origin-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('bg size', async ({ uno }) => {
  const classes = ['auto', 'cover', 'contain'].map(e => `bg-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('bg attachments', async ({ uno }) => {
  const classes = ['fixed', 'local', 'scroll'].map(e => `bg-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('bg clip', async ({ uno }) => {
  const staticClasses = ['bg-clip-border',
    'bg-clip-content',
    'bg-clip-padding',
    'bg-clip-text'];
  const globalKeyClasses = globalKeywords.map(keyword => `bg-clip-${keyword}`);
  const { css } = await uno.generate([...staticClasses, ...globalKeyClasses]);
  expect(css).toMatchSnapshot();
});

test('bg invalid', async ({ uno }) => {
  const classes = ['bg-white', 'bg-none'];
  const { css } = await uno.generate(classes);
  expect(css).toMatchInlineSnapshot('""');
});
