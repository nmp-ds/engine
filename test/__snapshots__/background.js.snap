// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bg attachments 1`] = `
"/* layer: default */
.bg-fixed{background-attachment:fixed;}
.bg-local{background-attachment:local;}
.bg-scroll{background-attachment:scroll;}"
`;

exports[`bg clip 1`] = `
"/* layer: default */
.bg-clip-border{-webkit-background-clip:border-box;background-clip:border-box;}
.bg-clip-content{-webkit-background-clip:content-box;background-clip:content-box;}
.bg-clip-padding{-webkit-background-clip:padding-box;background-clip:padding-box;}
.bg-clip-text{-webkit-background-clip:text;background-clip:text;}
.bg-clip-inherit{-webkit-background-clip:inherit;background-clip:inherit;}
.bg-clip-initial{-webkit-background-clip:initial;background-clip:initial;}
.bg-clip-revert{-webkit-background-clip:revert;background-clip:revert;}
.bg-clip-revert-layer{-webkit-background-clip:revert-layer;background-clip:revert-layer;}
.bg-clip-unset{-webkit-background-clip:unset;background-clip:unset;}"
`;

exports[`bg origins 1`] = `
"/* layer: default */
.bg-origin-border{background-origin:border-box;}
.bg-origin-padding{background-origin:padding-box;}
.bg-origin-content{background-origin:content-box;}"
`;

exports[`bg positions 1`] = `
"/* layer: default */
.bg-bottom{background-position:bottom;}
.bg-center{background-position:center;}
.bg-left{background-position:left;}
.bg-right{background-position:right;}
.bg-top{background-position:top;}
.bg-bottom-bottom{background-position:bottom bottom;}
.bg-bottom-center{background-position:bottom center;}
.bg-bottom-left{background-position:bottom left;}
.bg-bottom-right{background-position:bottom right;}
.bg-bottom-top{background-position:bottom top;}
.bg-center-bottom{background-position:center bottom;}
.bg-center-center{background-position:center center;}
.bg-center-left{background-position:center left;}
.bg-center-right{background-position:center right;}
.bg-center-top{background-position:center top;}
.bg-left-bottom{background-position:left bottom;}
.bg-left-center{background-position:left center;}
.bg-left-left{background-position:left left;}
.bg-left-right{background-position:left right;}
.bg-left-top{background-position:left top;}
.bg-right-bottom{background-position:right bottom;}
.bg-right-center{background-position:right center;}
.bg-right-left{background-position:right left;}
.bg-right-right{background-position:right right;}
.bg-right-top{background-position:right top;}
.bg-top-bottom{background-position:top bottom;}
.bg-top-center{background-position:top center;}
.bg-top-left{background-position:top left;}
.bg-top-right{background-position:top right;}
.bg-top-top{background-position:top top;}"
`;

exports[`bg repeats 1`] = `
"/* layer: default */
.bg-repeat{background-repeat:repeat;}
.bg-no-repeat{background-repeat:no-repeat;}
.bg-repeat-x{background-repeat:repeat-x;}
.bg-repeat-y{background-repeat:repeat-y;}
.bg-repeat-round{background-repeat:round;}
.bg-repeat-space{background-repeat:space;}"
`;

exports[`bg size 1`] = `
"/* layer: default */
.bg-auto{background-size:auto;}
.bg-cover{background-size:cover;}
.bg-contain{background-size:contain;}"
`;
